hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -312,6 +312,10 @@ def _load_dataset(self, slides_dataset: SlidesDataset, stage: ModelKey) -> Datas
             transforms = Compose([base_transform, self.transforms_dict[stage]]).flatten()
         else:
             transforms = base_transform
+        if self.seed is not None:
+            # The tiling transform is randomized. Make them deterministic. This call needs to be
+            # done on the final Compose, not at the level of the individual randomized transforms.
+            transforms.set_random_state(self.seed)
```

Comment:
I wonder if we still need to set_random_state for the previous  transform_train and transform_inf. They are going to be composed with the base transforms line 312 and what matters is to set_random_state on the final Compose line.318, is that right? so no need to fix the seed at the container level if it's controlled by the datamodule?
Note: This is even better because we control sources of randomness related to monai transforms with a single line of code in a much safer way (e.g. it's less likely to make changes in the base datamodule vs containers that change constantly during experiments)
Also no need to add extra comments in configs as I suggest it before if we are able to fix it at the base data module level.

Reply:
Very good point indeed. I've simplified the code